src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -648,40 +655,25 @@ public GitHub getGitHub() {
       return gitHub;
     }
 
-    @CheckForNull
-    private static GitHubConnection lookup(@NonNull ConnectionId connectionId) throws IOException {
-      GitHubConnection record;
-      record = connections.get(connectionId);
-      if (record != null) {
-        record.verifyConnection();
-        record.usageCount += 1;
-        record.lastUsed = System.currentTimeMillis();
-      }
-      return record;
-    }
-
     @NonNull
-    private static GitHubConnection connect(
+    private static GitHubConnection lookup(
         @NonNull ConnectionId connectionId,
-        @NonNull GitHub gitHub,
-        @CheckForNull Cache cache,
-        boolean cleanupCacheFolder)
-        throws IOException {
-      GitHubConnection record = new GitHubConnection(gitHub, cache, cleanupCacheFolder);
-      record.verifyConnection();
-      connections.put(connectionId, record);
-      reverseLookup.put(record.gitHub, record);
+        @NonNull Function<ConnectionId, GitHubConnection> generator) {
+      GitHubConnection record;
+      record = connections.computeIfAbsent(connectionId, generator);
+      record.usageCount.incrementAndGet();
+      record.lastUsed.compareAndSet(record.lastUsed.get(), System.currentTimeMillis());
```

Comment:
@timja
You didn't use `set()` here.  Is this done so, only one caller updates the value if there's race?

Reply:
yes that's why i did it that way